### PR TITLE
Tests for rotate_grid_vectors

### PR DESCRIPTION
**Pull request by pp-mo**
Previously deferred work, as noted in #3114

### CODE REVIEW

File: lib/iris/tests/unit/analysis/cartography/test_rotate_grid_vectors.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        self.assertTrue(np.ma.isMaskedArray(out_v))
+        if nan_angles_mask is not None:
+            self.assertArrayEqual(out_u.mask, nan_angles_mask)
+            self.assertArrayEqual(out_v.mask, nan_angles_mask)
+
+    def test_angles_calculation(self):
+        self._check_angles_calculation()
+
+    def test_angles_in_radians(self):
+        self._check_angles_calculation(angles_in_degrees=True)
```
Comment (lbdreyer) [inline lib/iris/tests/unit/analysis/cartography/test_rotate_grid_vectors.py]: This test is called `test_angles_in_radians` but it uses `angles_in_degrees=True`

File: lib/iris/tests/unit/analysis/cartography/test_rotate_grid_vectors.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+# importing anything else.
+import iris.tests as tests
+
+from mock import Mock, call as mock_call
+import numpy as np
+
+from iris.cube import Cube
+
+from iris.analysis.cartography import rotate_grid_vectors
+from iris.tests.stock import sample_2d_latlons
```
[the remark this answers is not in the document]
Comment (pp-mo) [reply lib/iris/tests/unit/analysis/cartography/test_rotate_grid_vectors.py]: The last bit is conventionally separate as it is the "Device Under Test"
But you're right, that the `sample_2d_latlons` needs to be integrated with the rest of the 'supporting' imports.  Will fix ..

File: lib/iris/tests/unit/analysis/cartography/test_rotate_grid_vectors.py (hunk 1 above)
Comment (pp-mo) [reply lib/iris/tests/unit/analysis/cartography/test_rotate_grid_vectors.py]: Whoops!  
Should obviously be 'False'. Will fix.

**Comment by pp-mo** [comment]
Thanks @lbdreyer 
I hope we're done here now ?

I'm not self-merging, as there's no immediate need (it's only extra tests really)

**Pull request merged**